### ISSUE TITLE
Loss is not decreasing, Acc not increasing in SSL_Backbone

### ISSUE THREAD

**Issue by timothyvanco** [larsh0103/SSL_Experiement#1]
Hi, First thank you for your code, I am learning more about self-supervised learning so I thought, this could be interesting, but I got stuck.

I tried your code but in training the first NN (make_SSL-Backbone part), the loss is not decreasing and acc is not increasing. Did you have the same problem? Or do you know, where could be a bug? Thanks a lot.

**Comment by larsh0103** [comment]
> Hi, First thank you for your code, I am learning more about self-supervised learning so I thought, this could be interesting, but I got stuck.
> 
> I tried your code but in training the first NN (make_SSL-Backbone part), the loss is not decreasing and acc is not increasing. Did you have the same problem? Or do you know, where could be a bug? Thanks a lot.

Hey, what platform are you working on, are you using google colab? I'll give it a run-through and see if I can reproduce this issue.

**Comment by larsh0103** [comment]
This bug is now fixed, thanks for bringing it to my attention! It was a bug in the make_rotated_data function. The images were being subtracted from themselves, instead of subtracting the pixel mean, thereby putting all pixels to 0, no wonder your NN wasn't learning anything.

**Comment by timothyvanco** [comment]
thanks a lot! so did you commit your change in a code?

**Comment by timothyvanco** [comment]
Hi, I would like to ask, for how long the code will run, because it is training on "functional_23" and still running

**Comment by larsh0103** [comment]
> Hi, I would like to ask, for how long the code will run, because it is training on "functional_23" and still running

Not sure what you are referring to, which training script are you talking about? If you are running this using GPU-acceleration the whole notebook should not take more than a few hours, with the default epoch and k-fold split configuration. You can reduce the number of epochs or k folds to speed up the training.
